Update contribute-upstream.md for modern Go workflow (#56490)
kubernetes/website: content/en/docs/contribute/generate-ref-docs/contribute-upstream.md

- You need to have these tools installed:

  - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)

>   - Go (use the version required by the `reference-docs` repository; see its `go.mod`)

Reviewer: This should just state "latest" or something. I don't think we want to send readers off to track down a different repo to read its go.mod file.

Author: Thanks for the feedback! Good point. I'll update this to avoid requiring readers to inspect another repository.

  - [Docker](https://docs.docker.com/engine/installation/)
  - [etcd](https://github.com/coreos/etcd/)
  - [make](https://www.gnu.org/software/make/)
  - [GCC compiler/linker](https://gcc.gnu.org/)

  - Ensure the `etcd` executable is available in your `PATH`.

> - Go modules are used to manage dependencies. You do not need to configure

Reviewer: I don't think this bullet is really necessary.

Author: Thanks! I agree this is redundant now that the guide no longer uses the GOPATH workflow. I've removed this bullet.

- You need to have these tools installed:

  - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
  - The latest stable version of [Go](https://go.dev/doc/install)
  - [Docker](https://docs.docker.com/engine/installation/)
  - [etcd](https://github.com/coreos/etcd/)
  - [make](https://www.gnu.org/software/make/)
  - [GCC compiler/linker](https://gcc.gnu.org/)

  - Ensure the `etcd` executable is available in your `PATH`.

  - You need to know how to create a pull request to a GitHub repository.
  Typically, this involves creating a fork of the repository.
  For more information, see [Creating a Pull Request](https://help.github.com/articles/creating-a-pull-request/)

>   and [GitHub Standard Fork & Pull Request Workflow](https://gist.github.com/Chaser324/ce0505fbed06b947d962).

Reviewer 2: Since we have an updated prerequisites page from PR #56403, we should used that instead

```suggestion
{{< include "prerequisites-ref-docs.md" >}}
```

Author: Thanks for the comments. I have updated the section.

## Clone the Kubernetes repository

> If you don't already have the 'kubernetes/kubernetes' repository, get it now:

Reviewer 2: nit: 

```suggestion
If you don't already have the kubernetes/kubernetes repository. You need a local clone.
```

Author: > nit:

I have applied the changes as requested. Thanks

> If you don't already have the 'kubernetes/kubernetes' repository, You need a local clone.

Reviewer 2: nit: use a backtick instead 

```suggestion
If you don't already have the `kubernetes/kubernetes` repository, You need a local clone.
```

Author: > nit: use a backtick instead

Updated to use backtick for the repository name. Thanks

> If you don't already have the `kubernetes/kubernetes` repository, You need a local clone.

Reviewer 2: Sorry, I think we should further improve here :

```suggestion
If you haven't already forked the `kubernetes/kubernetes` repository, first create a fork on GitHub. Then clone your fork locally:
```

Author: Addressed the comments.

```shell

> git clone https://github.com/kubernetes/kubernetes

Reviewer 2: ```suggestion
git clone https://github.com/<your-username>/kubernetes
```

Reviewer 2: And after this shell command, add this to tell readers what <k8s-base> mean for the remaining steps

```
The remaining steps refer to your `kubernetes/kubernetes` clone as `<k8s-base>`
```

Reviewer: You need to clone the `kubernetes/kubernetes` repo. This isn't showing in the diff view, so I'm not sure if I'm missing some context on this comment. But the change as it is appears to be correct.

Reviewer 2: Probably my suggestion added or need a refresh but the context is to let's contributor clone from their forked repo , (hence git clone https://github.com/<your-username>/kubernetes) and the last text ('The remaining steps refer to your `kubernetes/kubernetes` clone as `<k8s-base>`') which does not exist on the changes here yet is useful for the other part of the page that mention k8s-docs, like this part https://github.com/kubernetes/website/pull/56490/changes/BASE..4c8cb546cc9444dd96f64aea8dc90a00ece4dcb5#diff-c03f66245e25378b28933004de6119c8331b87ed42c50ef3b51ff307284309acL128

Author: > And after this shell command, add this to tell readers what mean for the remaining steps
> 
> ```
> The remaining steps refer to your `kubernetes/kubernetes` clone as `<k8s-base>`
> ```

Addressed the comment.

Author: > 

Addressed the comment.

Author: > You need to clone the `kubernetes/kubernetes` repo. This isn't showing in the diff view, so I'm not sure if I'm missing some context on this comment. But the change as it is appears to be correct.

Addressed the comment.

## Clone the Kubernetes repository

> Clone the upstream Kubernetes repository:

Reviewer 2: I think I prefer the old text

Author: Suggestion accepeted. Thanks.

```shell
git clone https://github.com/kubernetes/kubernetes
```
Throughout this guide:

> - `<k8s-base>` refers to your local clone of the `kubernetes/kubernetes` repository.

Reviewer 2: The old/existing content seems more beginner friendly. Consider reverting or improve but remove the GOPATH part.

Author: Thanks for the comments.

When you see bugs in the generated documentation, you may want to consider
creating a patch to fix it in the upstream project.

## Clone the repositories

Clone the repositories required for contributing to the generated reference documentation:

```shell
git clone https://github.com/kubernetes/kubernetes
git clone https://github.com/kubernetes-sigs/reference-docs

> git clone https://github.com/kubernetes/website

Reviewer: You don't really need these repos if you're just trying to contribute to kubernetes.

Author: Thanks for the comment. This has been taken care.

```

> You can clone these repositories into any workspace directory. They do not

Reviewer: For newer Go developers, these references to GOPATH might just be confusing. I think it's better to just remove the instructions that referred to the "old way" of doing things.

Author: Thanks for the feedback! I've removed the remaining GOPATH references from this guide and simplified the repository setup to use git clone with the local kubernetes/kubernetes clone referenced as <k8s-base>. Please let me know if I missed any remaining references.

```shell
cd <k8s-base>

> git checkout main

Reviewer 2: kubernetes is still using 'master'

Author: Thanks for the comment. Its done.

git status
```

The output shows that you are on the main branch, and that the `types.go`
source file has been modified:

```shell

> On branch main

Reviewer 2: kubernetes is still using 'master'

Author: Thanks for the comment. Its updated.

Run `git status` to see what was generated.

```shell

> On branch main

Reviewer 2: kubernetes is still using 'master'

Author: Thanks for the comment. Its updated.

Submit your changes as a
[pull request](https://help.github.com/articles/creating-a-pull-request/) to the

> main branch of the

Reviewer 2: kubernetes is still using 'master'

Author: Thanks for the comment. Updated.

> ### Cherry pick your commit into a release branch

Reviewer 2: see my comments here on this section

https://github.com/kubernetes/website/pull/56441/changes#r3566372837

Author: Thanks for the comment. Done.
